src/osd/PrimaryLogPG.cc

```diff
@@ -3457,24 +3458,121 @@ void PrimaryLogPG::cancel_manifest_ops(bool requeue, vector<ceph_tid_t> *tids)
   }
 }
 
-void PrimaryLogPG::refcount_manifest(ObjectContextRef obc, object_locator_t oloc, hobject_t soid,
-                                     SnapContext snapc, bool get, RefCountCallback *cb, uint64_t offset)
+void PrimaryLogPG::dec_refcount(ObjectContextRef obc, const object_info_t& oi, 
+				const object_ref_delta_t& refs)
+{
+  for (auto p = refs.begin(); p != refs.end(); ++p) {
```

Comment:
object_ref_delta_t is a mapping from the target to the ref delta.  The delta at the moment will always be negative because all users can only drop references (you can assert that).  However, the delta may be < -1, in which case you may need to drop more than one reference for a particular target.  p->second has the delta.

Reply:
When you fix this, please avoid rebasing or squashing so I can quickly find the fix and the test case covering it.